[22.x] build: Bump depends packages (zmq, libXau)

fanquake opened the pull request:
Backports #24212 to the 22.x branch. Included a bump to libXau and an additional commit for zeromq to fix dash compatibility.

fanquake committed:
build: libXau 1.0.9

Github-Pull: #23489
Rebased-From: fc65127244320500022c3772141850eda28b99af

Commit:
ci: Bump CentOS 8 image

Github-Pull: #24212
Rebased-From: fa5457e64adfc6060382bd36c45e95be6ea693ef

Commit:
ci: Use dash when building depends in centos build

Github-Pull: #24212
Rebased-From: fafc55a489d8f654dd7732b699c9bf51b5136b4b

hebasto commented:
Looks like libXau 1.0.8 couldn't be built with `dash`. Weird...

hebasto commented:
@fanquake 

This branch works: https://github.com/hebasto/bitcoin/commits/pr24256-22.x/fixed

fanquake committed:
build: fix depends zeromq dash compatibility

Picks upstream commit 9a397666d28ca5f3c0d8233be3d39b2206555f45 to fix
dash compatibility. This fixes building zeromq in our CentOS 8 CI. This
is my preferred fix over backporting a zeromq update (which would
contain this change).

MarcoFalke commented:
Changed title

fanquake commented:
> This branch works: https://github.com/hebasto/bitcoin/commits/pr24256-22.x/fixed

I would rather not backport a full zmq bump to a release branch. Should have resolved this be just picking the relevant commit from upstream: https://github.com/zeromq/libzmq/pull/3518/commits/9a397666d28ca5f3c0d8233be3d39b2206555f45. Also rebased for #24215.

MarcoFalke commented:
An alternative would be to simply not support `dash` and require CentOS 8 users to use `bash`

hebasto commented on depends/packages/zeromq.mk:
```diff
@@ -3,7 +3,7 @@ $(package)_version=4.3.1
 $(package)_file_name=$(package)-$($(package)_version).tar.gz
 $(package)_sha256_hash=bcbabe1e2c7d0eec4ed612e10b94b112dd5f06fcefa994a0c79a45d835cd21eb
-$(package)_patches=remove_libstd_link.patch
+$(package)_patches=remove_libstd_link.patch fix_dash_compat.patch
```
Apply patch in `$(package)_preprocess_cmds`?

fanquake replied:
Thanks. Adjusted the patch as well.

hebasto commented:
> An alternative would be to simply not support `dash` and require CentOS 8 users to use `bash`

Concept ACK on supporting dash in CentOS.

hebasto commented on depends/packages/zeromq.mk:
```diff
@@ -17,7 +17,9 @@ endef
-  cp -f $(BASEDIR)/config.guess $(BASEDIR)/config.sub config
+  patch -p1 < $($(package)_patch_dir)/fix_dash_compat.patch && \
+  cp -f $(BASEDIR)/config.guess $(BASEDIR)/config.sub config && \
+  ./autogen.sh
```
Why `./autogen.sh` is a part of `$(package)_preprocess_cmds`? On the master branch it belongs to `$(package)_config_cmds`:https://github.com/bitcoin/bitcoin/blob/3ce40e64d4ae9419658555fd1fb250b93f52684a/depends/packages/zeromq.mk#L27-L30

fanquake replied:
Changed.

hebasto commented on depends/patches/zeromq/fix_dash_compat.patch:
```diff
@@ -0,0 +1,71 @@
+commit 058158724b3003fa348a338a5b8f1d058dc59b99
```
Cannot find a commit with this hash.
Maybe https://github.com/zeromq/libzmq/commit/9a397666d28ca5f3c0d8233be3d39b2206555f45?

hebasto reviewed (approved):
ACK d34691ce0c07b5437c0e4fa9aea4afd699530151
